app/models/task.py

```diff
@@ -2,4 +2,35 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String(80))
+    description = db.Column(db.String(255))
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    @classmethod
+    def from_dict(cls, task_details):
+        new_task = cls(
+            title=task_details["title"],
+            description=task_details["description"],
+            completed_at=task_details.get("completed_at", None)
+        )
+
+        return new_task
```

Comment:
Beautiful class method! 🎉 

Small nit: remember to put a space before and after your `=`, as stated in the [pep8 documentation](https://peps.python.org/pep-0008/#blank-lines).